src/extension.ts

```diff
@@ -2,332 +2,529 @@ import { Luraph, type LuraphOptionList } from "luraph";
-			}
-		}catch(err){
-			if(err instanceof Error && err.name === "LuraphException"){ //TODO: use instanceof LuraphException
-				return vscode.window.showErrorMessage(`Luraph API Error: ${err.message}`);
-			}
-
-			throw err;
-		}finally{
-			statusBarItem.text = "$(terminal) Obfuscate with Luraph";
-		}
-	});
-
-	context.subscriptions.push(command);
+  const statusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+
+  statusBarItem.name = "Luraph";
+  statusBarItem.text = "$(terminal) Obfuscate with Luraph";
+  statusBarItem.tooltip = new vscode.MarkdownString(
+    `Obfuscate the currently opened file with [Luraph](https://lura.ph/ "Luraph - Online Lua Obfuscator").`
+  );
+  statusBarItem.command = "luraph.obfuscate";
+
+  statusBarItem.show();
+
+  const logOutput = vscode.window.createOutputChannel("Luraph", {
+    log: true,
+  });
+  const log = logOutput.info;
+
+  context.subscriptions.push(statusBarItem, logOutput); //auto dispose on extension deactivation
+
+  log("Luraph VS Code extension has started.");
+
+  const command = vscode.commands.registerCommand("luraph.obfuscate", async () => {
+    const settings = vscode.workspace.getConfiguration("luraph");
+    let apiKey: string | undefined = settings.get("API Key");
+
+    if (!apiKey?.length) {
+      const action = await vscode.window.showErrorMessage(
+        "An API key must be configured to use the Luraph API.",
+        {
+          title: "Set API Key",
+        }
+      );
+
+      if (!action) {
+        return;
+      }
+
+      const input = await vscode.window.showInputBox({
+        title: "Luraph - Set API Key",
+        prompt: "Please enter your Luraph API key.",
+        placeHolder: "Luraph API Key",
+
+        ignoreFocusOut: true,
+        validateInput: value => {
+          if (!value.length) {
+            return {
+              message: "API key must not be empty.",
+              severity: vscode.InputBoxValidationSeverity.Error,
+            };
+          }
+
+          return null;
+        },
+      });
+
+      if (!input) {
+        return;
+      }
+
+      apiKey = input;
+      settings.update("API Key", input, true); //update globally
+    }
+
+    const textEditor = vscode.window.activeTextEditor;
+    const document = textEditor?.document;
+
+    if (!document) {
+      return vscode.window.showErrorMessage("Please open an file to obfuscate.");
+    }
+
+    const fileName = document.fileName;
+    const contents = document.getText();
+
+    if (!contents.length) {
+      return vscode.window.showErrorMessage("Cannot obfuscate an empty file.");
+    }
+
+    log(`Performing Luraph obfuscation for ${fileName}...`);
+    const luraphApi = new Luraph(apiKey);
+
+    const availableNodes: vscode.QuickPickItem[] = [];
+
+    try {
+      // log("> Fetching nodes...");
+
+      const nodesPromise = luraphApi.getNodes();
+
+      vscode.window.withProgress(
```

Comment:
note to self: what does this do?

Reply:
Shows a pop up with loading indicator which closes when promise resolves